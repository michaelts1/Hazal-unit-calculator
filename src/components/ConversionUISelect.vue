<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
	props: {
		selectedUnit: {
			type: String,
			required: true,
		},
		unitNames: {
			/** @type {string[]} */
			type: Array,
			required: true,
		},
	},

	emits: ['value-change'],
})
</script>

<template>
	<select
		:value="selectedUnit"
		@change="({ target }) => $emit('value-change', target.value)"
	>
		<option
			v-for="name in unitNames"
			:key="name"
			:value="name"
		>
			{{ name }}
		</option>
	</select>
</template>
