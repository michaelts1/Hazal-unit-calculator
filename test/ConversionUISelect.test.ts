import { describe, expect, it } from 'vitest'
import ConversionUISelect from '../src/components/ConversionUISelect.vue'
import { mount } from '@vue/test-utils'

describe('Component ConversionUISelect', () => {
	it('exists', () => {
		expect(ConversionUISelect).toBeTruthy()
	})

	const unitNames = [ 'טפח', 'אצבע' ]
	const wrapper = mount(ConversionUISelect, {
		props: {
			selectedUnit: unitNames[0],
			unitNames,
		},
	})

	it('has a select field with the correct number of options', () => {
		expect(wrapper.find('select').exists()).toBe(true)
		expect(wrapper.findAll('option')).toHaveLength(unitNames.length)
	})

	it('matches snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('reacts to value changes', async () => {
		expect(wrapper.vm.selectedUnit).toBe(unitNames[0])

		await wrapper.find('select').setValue(unitNames[1])
		const changeEvents = wrapper.emitted('change')
		const valueChangeEvents = wrapper.emitted('value-change') as unknown[]

		expect(changeEvents).toHaveLength(1)
		expect(valueChangeEvents).toHaveLength(1)
		expect(valueChangeEvents[0]).toEqual([ unitNames[1] ])
	})
})
