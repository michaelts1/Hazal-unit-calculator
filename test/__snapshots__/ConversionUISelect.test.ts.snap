// Vitest Snapshot v1

exports[`Component ConversionUISelect > matches snapshot 1`] = `
"<select>
  <option value=\\"טפח\\">טפח</option>
  <option value=\\"אצבע\\">אצבע</option>
</select>"
`;
